src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQueryPushDataset.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.hadoop.io.bigquery.output.BigQueryTableFieldSchema;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.SerializableTransform;
+import io.cdap.cdap.etl.api.Transform;
+import io.cdap.cdap.etl.api.engine.sql.dataset.SQLPushDataset;
+import io.cdap.cdap.etl.api.engine.sql.request.SQLPushRequest;
+import io.cdap.plugin.gcp.bigquery.sink.BigQueryOutputFormatProvider;
+import io.cdap.plugin.gcp.bigquery.sink.BigQuerySinkUtils;
+import io.cdap.plugin.gcp.bigquery.sink.Operation;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryConstants;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * SQL Push Dataset implementation for BigQuery backed datasets.
+ */
+public class BigQueryPushDataset extends BigQueryOutputFormatProvider
+  implements SQLPushDataset<StructuredRecord, StructuredRecord, NullWritable>, BigQuerySQLDataset {
+
+  // Note that these properties are not final in order to support Spark's serialization.
```

Comment:
Does this class need to be serializable? I don't see anything here that would looks like Spark would need to serialize this.

Reply:
can remove this comment now